intel-hex.d.ts

```diff
@@ -0,0 +1,311 @@
+   * <tt>Uint8Array</tt>s.
+   *
+   * @return {MemoryMap}
+   */
+  clone(): MemoryMap;
+
+  /**
+   * Given one <tt>Uint8Array</tt>, looks through its contents and returns a new
+   * {@linkcode MemoryMap}, stripping away those regions where there are only
+   * padding bytes.
+   * <br/>
+   * The start of the input <tt>Uint8Array</tt> is assumed to be offset zero for the output.
+   * <br/>
+   * The use case here is dumping memory from a working device and try to see the
+   * "interesting" memory regions it has. This assumes that there is a constant,
+   * predefined padding byte value being used in the "non-interesting" regions.
+   * In other words: this will work as long as the dump comes from a flash memory
+   * which has been previously erased (thus <tt>0xFF</tt>s for padding), or from a
+   * previously blanked HDD (thus <tt>0x00</tt>s for padding).
+   * <br/>
+   * This method uses <tt>subarray</tt> on the input data, and thus does not allocate memory
+   * for the <tt>Uint8Array</tt>s.
+   *
+   * @param {Uint8Array} bytes The input data
+   * @param {Number} [padByte=0xFF] The value of the byte assumed to be used as padding
+   * @param {Number} [minPadLength=64] The minimum number of consecutive pad bytes to
+   * be considered actual padding
+   *
+   * @return {MemoryMap}
+   */
+  static fromPaddedUint8Array(
+    bytes: Uint8Array,
+    padByte?: number,
+    minPadLength?: number
+  ): MemoryMap;
+
+  /**
+   * Returns a new instance of {@linkcode MemoryMap}, containing only data between
+   * the addresses <tt>address</tt> and <tt>address + length</tt>.
+   * Behavior is similar to {@linkcode https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/slice|Array.prototype.slice},
+   * in that the return value is a portion of the current {@linkcode MemoryMap}.
+   *
+   * <br/>
+   * The returned {@linkcode MemoryMap} might be empty.
+   *
+   * <br/>
+   * Internally, this uses <tt>subarray</tt>, so new memory is not allocated.
+   *
+   * @param {Number} address The start address of the slice
+   * @param {Number} length The length of memory map to slice out
+   * @return {MemoryMap}
+   */
+  slice(address: number, length?: number): MemoryMap;
+
+  /**
+   * Returns a new instance of {@linkcode https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/getUint32|Uint8Array}, containing only data between
+   * the addresses <tt>address</tt> and <tt>address + length</tt>. Any byte without a value
+   * in the input {@linkcode MemoryMap} will have a value of <tt>padByte</tt>.
+   *
+   * <br/>
+   * This method allocates new memory.
+   *
+   * @param {Number} address The start address of the slice
+   * @param {Number} length The length of memory map to slice out
+   * @param {Number} [padByte=0xFF] The value of the byte assumed to be used as padding
+   * @return {MemoryMap}
+   */
+  slicePad(address: number, length: number, padByte?: number): MemoryMap;
```

Comment:
```suggestion
   * @return {Uint8Array}
   */
  slicePad(address: number, length: number, padByte?: number): Uint8Array;
```

Reply:
Yep. We should probably fix the function header in the source file to match this correct type.